src/swoop/api/main.py

```diff
@@ -14,6 +15,20 @@
 
 app: FastAPI = FastAPI()
 
+app.state.settings = settings
+
+@app.on_event("startup")
+async def startup_event():
+    """Connect to database on startup."""
+    pools = await connect_to_db(settings)
+    app.state.readpool = pools[0]
+    app.state.writepool = pools[1]
+
+@app.on_event("shutdown")
+async def shutdown_event():
+    """Close database connection."""
+    await close_db_connection(app.state)
```

Comment:
Just wondering a bit how this works? The open connections are stored in `app.state` in some way?

Reply:
Doh, I was being obtuse for a moment. I see it now.

Reply:
Yeah, stored in app.state.readpool / writepool. In the stac code, the DB class tucks those inside of app.state, but I didn't love that because I wanted to be able to write a test for the DB that doesn't require FastApi involvement.

https://github.com/stac-utils/stac-fastapi/blob/01bc0d58697743e870a768b0d486dde7bd567d9f/stac_fastapi/pgstac/stac_fastapi/pgstac/db.py#L37-L47